core/services/gateway/handlers/functions/allowlist.go

```diff
@@ -191,3 +232,31 @@ func (a *onchainAllowlist) update(addrList []common.Address) {
 	a.allowlist.Store(&newAllowlist)
 	a.lggr.Infow("allowlist updated successfully", "len", len(addrList))
 }
+
+func (a *onchainAllowlist) updateCache(addrList []common.Address) {
```

Comment:
I'm trying to understand... the `.update()` method stores the list in memory, while `updateCache()` stores it to disk.
Is that correct?

If so, nit that this is being called the cache.

Reply:
yeah its a bad naming indeed, because this is actually more a stored snapshot than a cache. Im going to change it here, then will need to change on the already merged subscriptions implementation, probably in a different mr